mobile/Screen/MapScreen.js

```diff
@@ -1,39 +1,265 @@
+      },
+      {
+        id: 2,
+        coordinate: {
+          latitude: 34.046070,
+          longitude: -118.247540,
+        },
+        title: 'Open Mic Night with Unkal Bean (King Eddy Saloon)',
+        description: '131 E 5th St.',
+        image: Images[1],
+      },
+      {
+        id: 3,
+        coordinate: {
+          latitude: 34.043580,
+          longitude: -118.247680,
+        },
+        title: 'Piece by Piece (Star Apartments)',
+        description: '240 E 6th St. | (323) 963-3372',
+        image: Images[2],
+      },
+      {
+        id: 4,
+        coordinate: {
+          latitude: 34.044536,
+          longitude: -118.244873,
+        },
+        title: 'Movies on the Nickel (James Wood Community Center',
+        description: '400 E 5th St. | (213) 229-9602',
+        image: Images[3],
+      },
+    ],
+    region: {
+      latitude: 34.0442,
+      longitude: -118.2439,
+      latitudeDelta: 0.005, // smaller value = more zoomed in
+      longitudeDelta: 0.005,
+    },
   };
 
+  const [mounted, setMounted] = useState(false);
+
+  if (!mounted) {
+    mapRef.index = 0;
+    mapRef.animation = new Animated.Value(0);
+  }
+  useEffect(() => {
+    // We should detect when scrolling has stopped then animate
+    // We should just debounce the event listener here
+    mapRef.animation.addListener(({ value }) => {
+      let index = Math.floor(value / CARD_WIDTH + 0.3);
+      if (index >= state.markers.length) {
+        index = state.markers.length - 1;
+      }
+      if (index <= 0) {
+        index = 0;
+      }
+
+      clearTimeout(mapRef.regionTimeout);
+      mapRef.regionTimeout = setTimeout(() => {
+        if (mapRef.index !== index) {
+          mapRef.index = index;
+          const { coordinate } = state.markers[index];
+          mapRef.current.animateToRegion(
+            {
+              ...coordinate,
+              latitudeDelta: state.region.latitudeDelta,
+              longitudeDelta: state.region.longitudeDelta,
+            },
+            350,
+          );
+        }
+      }, 10);
+    });
+  }, []);
+
+  const interpolations = state.markers.map((marker, index) => {
+    const inputRange = [
+      (index - 1) * CARD_WIDTH,
+      index * CARD_WIDTH,
+      (index + 1) * CARD_WIDTH,
+    ];
+    const scale = mapRef.animation.interpolate({
+      inputRange,
+      outputRange: [1, 2.5, 1],
+      extrapolate: 'clamp',
+    });
+    const opacity = mapRef.animation.interpolate({
+      inputRange,
+      outputRange: [0.35, 1, 0.35],
+      extrapolate: 'clamp',
+    });
+    return { scale, opacity };
+  });
+
+  const markers = state.markers.map((marker, index) => {
+    const opacityStyle = {
+      opacity: interpolations[index].opacity,
+    };
+    return (
+      <Marker key={marker.id} coordinate={marker.coordinate}>
+        <Animated.View style={[styles.markerWrap, opacityStyle]}>
+          <View style={styles.marker} />
+        </Animated.View>
+      </Marker>
+    );
```

Comment:
This is really cool but it might be better to create a separate component in a different file that way you don't need to go searching to adjust the marker